token/cli/src/main.rs

```diff
@@ -1296,8 +1296,18 @@ fn command_gc(config: &Config, owner: Pubkey) -> CommandResult {
 
         for (address, (amount, decimals, frozen, close_authority)) in accounts {
             if address == associated_token_account {
-                // leave the associated token account alone
-                continue;
+                if !del_associated_accounts {
+                    // leave the associated token account alone
+                    continue;
+                } else {
+                    // close non empty associated token accounts
+                    if amount > 0 {
+                        println_display(config, format!("Not closing account {}. Balance > 0", associated_token_account));
+                        continue;
+                    } else {
+                        println_display(config, format!("Closing Account {}", associated_token_account));
+                    }
+                }
```

Comment:
nit: since there's lots possible combinations, a match statement may be easier to reason, ie something like:
```
match (address == associated_token_account, close_empty_associated_accounts, amount > 0) {
    (true, _, true) => continue, // don't ever close associated token account with amount
    (true, false, _) => continue, // don't close associated token account if close_empty_associated_accounts isn't set
    _ => {},
}
```